src/components/HSButton/HSButton.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import '../../styles/index.scss';
+
+import { ButtonVariant, HSButtonProps } from './types';
+
+const HSButton: React.FC<HSButtonProps> = ({
+  label = 'Click Me!',
+  icon,
+  type = null,
+  color = null,
```

Comment:
I think we should only limit the color to "success", "warning", and "danger". Then call the prop "state" to be in line with the [button specifications](https://hashi-docs.netlify.app/docs/next/design/button#states) in the docs.